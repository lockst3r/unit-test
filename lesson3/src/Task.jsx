import React from 'react';
import PropTypes from 'prop-types';

const Task = ({id, done, text, onChange, onDelete}) => {
    return(
        <li className={`list-item${done ? ' list-item_done': ''}`}>
            <input className="list-item__checkbox"
                type="checkbox" 
                checked={done}
                onChange={() => onChange(id)}
            />
            <span className="list-item__text">
                {text}
            </span>
            <button className="list-item__delete-btn"
                onClick={() => onDelete(id)}
            ></button> 
        </li>
    );
}

Task.propTypes = {
    id: PropTypes.string.isRequired,
    text: PropTypes.string,
    done: PropTypes.bool,
    onChange: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired,
};

Task.defaultProps = {
    done: false,
    text: '',
};

export default Task;