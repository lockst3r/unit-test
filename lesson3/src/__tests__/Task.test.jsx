import React from "react";
import { shallow } from "enzyme";
import Task from "../Task";

describe("<Task />", () => {
  it("Should display done task", () => {
    const props = {
      text: "text",
      id: "some-id-1",
      done: true,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    };
    const wrapperComponent = shallow(<Task {...props} />);
    expect(wrapperComponent).toMatchSnapshot();
  });

  it("Should display undone task", () => {
    const props = {
      text: "text",
      id: "some-id-1",
      done: false,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    };
    const wrapperComponent = shallow(<Task {...props} />);
    expect(wrapperComponent).toMatchSnapshot();
  });

  it("Should update event on checkbox cheked", () => {
    const props = {
      text: "text",
      id: "some-id-1",
      done: false,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    };
    const wrapperComponent = shallow(<Task {...props} />);
    wrapperComponent.find(".list-item__checkbox").simulate("change");
    expect(props.onChange).toBeCalledWith("some-id-1");
  });

  it("Should delete task", () => {
    const props = {
      text: "text",
      id: "some-id-1",
      done: false,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    };
    const wrapperComponent = shallow(<Task {...props} />);
    wrapperComponent.find(".list-item__delete-btn").simulate("click");
    expect(props.onDelete).toBeCalledWith("some-id-1");
  });
});
