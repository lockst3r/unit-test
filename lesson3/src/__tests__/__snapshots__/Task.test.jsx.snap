// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> Should display done task 1`] = `
<li
  className="list-item list-item_done"
>
  <input
    checked={true}
    className="list-item__checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    text
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;

exports[`<Task /> Should display undone task 1`] = `
<li
  className="list-item"
>
  <input
    checked={false}
    className="list-item__checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    text
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;
